src/backend/utils.ts

```diff
@@ -1483,66 +1478,6 @@ function calculateEta(
   return eta
 }
 
-interface ExtractOptions {
-  path: string
-  destination: string
-  strip: number
-}
-
-async function extractFiles({ path, destination, strip = 0 }: ExtractOptions) {
-  if (!isSnap && (path.endsWith('.tar.xz') || path.endsWith('.tar.gz'))) {
-    try {
-      await extractNative(path, destination, strip)
-    } catch (error) {
-      logError(['Error:', error], LogPrefix.Backend)
-    }
-  } else {
-    try {
-      await extractDecompress(path, destination, strip)
-    } catch (error) {
-      logError(['Error:', error], LogPrefix.Backend)
-    }
-  }
-}
-
-async function extractNative(path: string, destination: string, strip: number) {
-  logInfo(
-    `Extracting ${path} to ${destination} using native tar`,
-    LogPrefix.Backend
-  )
-  const { code, stderr } = await spawnAsync('tar', [
-    '-xf',
-    path,
-    '-C',
-    destination,
-    `--strip-components=${strip}`
-  ])
-  if (code !== 0) {
-    logError(`Extracting Error: ${stderr}`, LogPrefix.Backend)
-    return { status: 'error', error: stderr }
-  }
-  return { status: 'done', installPath: destination }
-}
```

Comment:
This function should stay & become the new `extractFiles` IMO

Doing it that way would mean you wouldn't need all the changes in `backend/tools/index.ts`